akka-docs/src/test/java/jdocs/actor/typed/BlockingFutureActor.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) 2009-2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package jdocs.actor.typed;
+
+import akka.actor.typed.*;
+import akka.actor.typed.javadsl.*;
+import akka.dispatch.Futures;
+import scala.concurrent.ExecutionContext;
+import scala.concurrent.Future;
+
+// #blocking-in-future
+class BlockingFutureActor extends AbstractBehavior<Integer> {
+  private final ExecutionContext ec;
+
+  public static Behavior<Integer> create() {
+    return Behaviors.setup(BlockingFutureActor::new);
+  }
+
+  private BlockingFutureActor(ActorContext<Integer> context) {
+    ec = context.getExecutionContext();
+  }
+
+  @Override
+  public Receive createReceive() {
+    return newReceiveBuilder()
+        .onMessage(
+            Integer.class,
+            i -> {
+              System.out.println("Calling blocking Future: " + i);
+              Future<Integer> f =
+                  Futures.future(
+                      () -> {
+                        Thread.sleep(5000);
+                        System.out.println("Blocking future finished: " + i);
+                        return i;
+                      },
```

Comment:
break out this to a `static` method, so that Future code is not written inside the actor instance

(at some point we should add that advise somewhere in docs)

Reply:
BTW, that should be CompletionStage instead of Future. maybe `CompletableFuture.supplyAsync`

Reply:
:+1: will break this out into a static method.

This section is about putting blocking calls on a separate dispatcher. I wonder if we should show `CompletableFuture` examples instead of `Future` examples, or as well as. Will look into it in any case.

Reply:
since we have moved away from using Future in javadsl everywhere we should not use it in Java examples. Could be a separate ticket if you prefer.

Reply:
It'll need some tweaks to the wording as well so I'll do that in a subsequent PR :+1: